sat/apiclient/vcs.py

```diff
@@ -75,38 +140,10 @@ def remote_refs(self):
             VCSError: if there is an error when git accesses VCS to enumerate
                 remote refs
         """
-
-        user_netloc = f'{self.username}@{self.vcs_host}'
-        url_components = ParseResult(scheme='https', netloc=user_netloc, path=self.repo_path,
-                                     params='', query='', fragment='')
-        vcs_full_url = urlunparse(url_components)
-
-        # Get the password directly from k8s to avoid leaking it via the /proc
-        # filesystem.
-        # TODO (CRAYSAT-1214): cfs-config-util uses the Python kubernetes
-        # library (via a small `vcs-creds-helper` script) since kubectl isn't
-        # available in the cfs-config-util image. As part of CRAYSAT-1214, we
-        # probably want to figure out a consistent way to retrieve credentials
-        # regardless of whether kubectl is installed. An argument could be made
-        # that both approaches used are clunky in their own way, and perhaps
-        # there's a cleaner common solution for both.
-        with NamedTemporaryFile(delete=False) as cred_helper_script:
-            cred_helper_script.write('#!/bin/bash\n'
-                                     'kubectl get secret -n services vcs-user-credentials '
-                                     '--template={{.data.vcs_password}} | base64 -d'.encode())
-
-        os.chmod(cred_helper_script.name, 0o700)  # Make executable by user
-
-        env = dict(os.environ)
-        env.update(GIT_ASKPASS=cred_helper_script.name)
         try:
-            proc = subprocess.run(['git', 'ls-remote', vcs_full_url],
-                                  stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                                  env=env, check=True)
+            proc = self.run_authenticated_git_cmd(['git', 'ls-remote', self.user_clone_url])
```

Comment:
I agree with you that we wouldn't have a way to handle authentication if given a different username than what is specified in the `vcs-user-credentials` K8s secret.

As a result, I've added code to `VCSRepo.__init__` that will check whether a username is specified in the `netloc` of the `clone_url`, and if so, it will discard the username and log a warning message to indicate it is being ignored. I think it would also make sense to do this validation as an earlier step when processing the CFS configuration layers, so we could be really clear about which layer in the input file has the problem, but this will be easier to do as a part of the refactor in [CRAYSAT-1278](https://jira-pro.its.hpecorp.net:8443/browse/CRAYSAT-1278). I added a comment there as a reminder.